base/pmap.jl

```diff
@@ -15,38 +15,81 @@ Note that `f` must be made available to all worker processes; see
 and Loading Packages <man-parallel-computing-code-availability>`)
 for details.
 """
-function pgenerate(p::WorkerPool, f, c)
-    if length(p) == 0
-        return AsyncGenerator(f, c)
+function pgenerate(p::WorkerPool, f, c; config=DEFAULT_PMAP_ARGS)
+    batch_size = config[:batch_size]
+    on_error = config[:on_error]
+    distributed = config[:distributed]
+
+    if  (distributed == false) ||
+        (length(p) == 0) ||
+        (length(p) == 1 && fetch(p.channel) == myid())
+
+        return AsyncGenerator(f, c; on_error=on_error)
+    end
+
+    if batch_size == :auto
+        batches = batchsplit(c, min_batch_count = length(p) * 3)
+    else
+        batches = batchsplit(c, max_batch_size = batch_size)
     end
-    batches = batchsplit(c, min_batch_count = length(p) * 3)
-    return flatten(AsyncGenerator(remote(p, b -> asyncmap(f, b)), batches))
+    return flatten(AsyncGenerator(remote(p, b -> asyncmap(f, b; on_error=on_error)), batches; on_error=on_error))
 end
 
-pgenerate(p::WorkerPool, f, c1, c...) = pgenerate(p, a->f(a...), zip(c1, c...))
+pgenerate(p::WorkerPool, f, c1, c...; kwargs...) = pgenerate(p, a->f(a...), zip(c1, c...); kwargs...)
 
-pgenerate(f, c) = pgenerate(default_worker_pool(), f, c...)
-pgenerate(f, c1, c...) = pgenerate(a->f(a...), zip(c1, c...))
+pgenerate(f, c; kwargs...) = pgenerate(default_worker_pool(), f, c...; kwargs...)
+pgenerate(f, c1, c...; kwargs...) = pgenerate(a->f(a...), zip(c1, c...); kwargs...)
 
 
 """
-    pmap([::WorkerPool], f, c...) -> collection
+    pmap([::WorkerPool], f, c...; distributed=true, batch_size=1, on_error=nothing) -> collection
 
 Transform collection `c` by applying `f` to each element using available
 workers and tasks.
 
 For multiple collection arguments, apply f elementwise.
 
-Note that `err_retry=true` and `err_stop=false` are deprecated,
-use `pmap(retry(f), c)` or `pmap(@catch(f), c)` instead
-(or to retry on a different worker, use `asyncmap(retry(remote(f)), c)`).
-
 Note that `f` must be made available to all worker processes; see
 [Code Availability and Loading Packages](:ref:`Code Availability
 and Loading Packages <man-parallel-computing-code-availability>`)
 for details.
+
+If a worker pool is not specified, all available workers, i.e., the default worker pool
+is used.
+
+By default, `pmap` distributes the computation over all specified workers. To use only the
+local process and distribute over tasks, specifiy `distributed=false`
```

Comment:
Maybe we should point out that this is equivalent to `asyncmap`. Some people may prefer to write `asyncmap(f, c)` instead of `pmap(f, c; distributed=false)`.
http://docs.julialang.org/en/latest/stdlib/parallel/?highlight=procs#Base.asyncmap